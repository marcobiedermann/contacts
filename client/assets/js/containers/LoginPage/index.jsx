import { firestoreConnect } from 'react-redux-firebase';
import { compose } from 'redux';
import LoginPage from '../../pages/LoginPage';

export default compose(firestoreConnect(() => ['contacts']))(LoginPage);
