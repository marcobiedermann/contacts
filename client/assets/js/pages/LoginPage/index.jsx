import { ErrorMessage, Field, Form, Formik } from 'formik';
import PropTypes from 'prop-types';
import React from 'react';
import Label from '../../components/Label';

const LoginPage = props => {
  const { firebase } = props;

  return (
    <div>
      <Formik
        initialValues={{
          email: '',
          password: '',
        }}
        onSubmit={async (values, actions) => {
          const { email, password } = values;

          try {
            const response = await firebase.login({
              email,
              password,
            });
            console.log(response);
          } catch (error) {
            console.error(error);
          }

          actions.setSubmitting(false);
        }}
        render={props => {
          const { isSubmitting } = props;

          return (
            <Form>
              <div>
                <Label htmlFor="email">Email</Label>
                <Field type="email" name="email" />
                <ErrorMessage name="email" />
              </div>

              <div>
                <Label htmlFor="password">Password</Label>
                <Field type="password" name="password" />
                <ErrorMessage name="password" />
              </div>

              <div>
                <input type="submit" disabled={isSubmitting} />
              </div>
            </Form>
          );
        }}
      />
      <button onClick={() => firebase.login({ provider: 'github', type: 'popup' })} type="button">
        Login with GitHub
      </button>
    </div>
  );
};

LoginPage.propTypes = {
  firebase: PropTypes.shape(),
  isSubmitting: PropTypes.bool,
};

LoginPage.defaultProps = {
  firebase: null,
  isSubmitting: false,
};

export default LoginPage;
