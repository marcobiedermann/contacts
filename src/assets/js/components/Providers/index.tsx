import React, { FC, Suspense } from 'react';
import { FirebaseAppProvider } from 'reactfire';
import config from '../../constants/config';
import Loader from '../Loader';

const Providers: FC = (props) => {
  const { children } = props;
  return (
    <FirebaseAppProvider firebaseConfig={config.firebase} suspense>
      <Suspense fallback={<Loader />}>{children}</Suspense>
    </FirebaseAppProvider>
  );
};

export default Providers;
