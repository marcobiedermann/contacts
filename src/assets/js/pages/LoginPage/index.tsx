import React, { FC } from 'react';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import { RouteComponentProps } from 'react-router-dom';
import { useAuth } from 'reactfire';

const LoginPage: FC<RouteComponentProps> = () => {
  const auth = useAuth;

  const uiConfig = {
    signInFlow: 'popup',
    signInOptions: [auth.GoogleAuthProvider.PROVIDER_ID],
    callbacks: {
      signInSuccessWithAuthResult: () => false,
    },
  };

  return (
    <div>
      <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={auth()} />
      {/* <Formik
        initialValues={{
          email: '',
          password: '',
        }}
        onSubmit={async (values, actions) => {
          const { email, password } = values;

          try {
            const response = await firebase.login({
              email,
              password,
            });
            console.log(response);
          } catch (error) {
            console.error(error);
          }

          actions.setSubmitting(false);
        }}
        render={(props) => {
          const { isSubmitting } = props;

          return (
            <Form>
              <div>
                <Label htmlFor="email">Email</Label>
                <Field type="email" name="email" />
                <ErrorMessage name="email" />
              </div>

              <div>
                <Label htmlFor="password">Password</Label>
                <Field type="password" name="password" />
                <ErrorMessage name="password" />
              </div>

              <div>
                <input type="submit" disabled={isSubmitting} />
              </div>
            </Form>
          );
        }}
      />
      <button onClick={() => firebase.login({ provider: 'github', type: 'popup' })} type="button">
        Login with GitHub
      </button> */}
    </div>
  );
};

export default LoginPage;
