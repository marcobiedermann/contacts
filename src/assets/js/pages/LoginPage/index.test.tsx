import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import LoginPage from '.';
import Providers from '../../components/Providers';

describe('login page component', () => {
  it('should redirect to `/`', () => {
    expect.assertions(1);

    const { asFragment } = render(
      <Providers>
        <MemoryRouter>
          <Route component={LoginPage} />
        </MemoryRouter>
      </Providers>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
