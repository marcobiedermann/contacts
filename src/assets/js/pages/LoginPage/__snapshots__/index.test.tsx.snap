// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page component should redirect to \`/\` 1`] = `
<DocumentFragment>
  <div>
    loading â€¦
  </div>
</DocumentFragment>
`;
